### ISSUE TITLE
Brainstorming on the features for the website

### ISSUE THREAD

**Issue by ipsha21** [ipsha21/altruism-in-tech#9]
A brief list of features that the website would primarily involve:

- Registration for mentors
- Registration for newcomers
- Displaying a database of registered mentors 
- Searching the database with keywords or tags
- Newcomer requesting a meet, at some mentor's profile
- Mentor reverting to the request 
- Setting up a meet, if mutually accepted by both parties
- Platform for ask-to-answer? 

Feel free to suggest changes, improvements, or more ideas for features to be included in the website.
Thanks! 😄

**Comment by share-with-me** [comment]
List of some amazing features indeed :D. I am myself working on a set of features that we could integrate with the above. However, from the above list, I have few questions:

> Setting up a meet, if mutually accepted by both parties

This meeting is online right (seems obvious but still :P)? 

> Platform for ask-to-answer?

Something like stackoverflow? I myself planned to integrate this feature. My vote definitely in favour!

> Displaying a database of registered mentors

Will this get too exhaustive with the number of mentors that will register on the website (are we looking at pagination?) ? Also, what will this database table hold? Mentor details and their field of expertise (and..) perhaps a button in the database itself that will take to the profile page of the mentor? 

> Searching the database with keywords or tags

Yep, feature will greatly enhance with this functionality.

In addition to the above, a few features that I have in mind:

- Personal profile page for every account, be it new comer or the mentor. This page will show the details of the person - name, educational background, field of expertise , experience, projects, etc. (can be thought of in the long run). A person could simply visit any profile to know about the mentor/new comer.

- For each account, a tab which lists the pending meetings, current status of the task (that the parties were discussing), and ability to add reminders, integrate calendars, cancel or setup new meetings, etc. (to be thought in detail)

- Perhaps a feature where the mentor and mentee could share screen and virtual text editor and code editor that would make it very convenient for the mentors to explain.

- A chat functionality to enable discussions between the mentor and mentee. To avoid spamming, we could embed a few things like, enabling chat only between the assigned mentor; or providing the feature where the mentor could make himself available (a status to others saying that the mentor is online and ready to guide) and only then, the chat can be done.

- A functionality (extended chat) to enable audio/video calls between mentor and mentee (at their will obviously).

I would keep on adding to this list as and when something pops in my mind.

**Comment by pransh15** [comment]
> This meeting is online right (seems obvious but still :P)?

Yes, obviously!

> Something like stackoverflow? I myself planned to integrate this feature. My vote definitely in favour!

Yes, something like stackoverflow!

> Will this get too exhaustive with the number of mentors that will register on the website (are we looking at pagination?) ? Also, what will this database table hold? Mentor details and their field of expertise (and..) perhaps a button in the database itself that will take to the profile page of the mentor?

We can section them out according to their field of expertise so that right from the landing page, would the mentee have an idea about where to go to learn say, Ruby on Rails and the Mentors would be listed accordingly.
Mentor Details, Field of Expertise, Previous Projects, A brief welcoming intro, Favorite Food(Cause, why not!?), suggest some more.

> Yep, feature will greatly enhance with this functionality.

Right on the landing page, the Mentee could go Python and _whoosh_, the Mentors available would show up, is my idea. What say?

The other features you mentioned, great! I'll add some features if something hits me.

**Comment by ipsha21** [comment]
Hi @share-with-me, sorry for such a late response.
@pransh15 cleared covered most of it (thanks!), but just a quick walk through your pointers:

> What will this database table hold?

An optional picture, name, affiliation, location. Clicking on the name/picture should land us on their profile, which brings me to the next point that you mentioned - _an individual profile page for each mentor_. We can discuss things to be included here: https://github.com/ipsha21/altruism-in-tech/issues/12

> For each account, a tab which lists the pending meetings, current status of the task (that the parties were discussing), and ability to add reminders, integrate calendars, cancel or setup new meetings, etc. (to be thought in detail)

That's a brilliant addition on what I missed out, thanks. :) Would love to discuss this in detail, feel free to open an issue about this. :)

> Perhaps a feature where the mentor and mentee could share screen and virtual text editor and code editor that would make it very convenient for the mentors to explain.

I had a more general idea in my mind, of only setting up the meeting (without really going into the execution part of how the meeting would be conducted - given that we already have loads of applications for that and users can choose with more flexibility, depending on their needs and ease), but I'm still open to the idea and we can work something out. What do you think? 

> A functionality (extended chat) to enable audio/video calls between mentor and mentee (at their will obviously).

Again, more in favor of the existing solutions like Skype or Hangouts. Opinions?

**Comment by amirad** [comment]
I was thinking, is there a rating process or something we can do? Or surveys. 

So if you meet a mentor/mentee you can say how they helped you which would be valuable for others to see. Something that is engaging and boosts mentors profiles over time. Ex. Leave a quote or testimonial for the mentor!

**Comment by ipsha21** [comment]
@amirad Yes, we could definitely set up some sort of recommendation/endorsement feature where mentees can write a testimonial for the mentor's profile. Also, a count of number of successful calls the mentor has been a part of.

Thanks for suggesting this. :)

**Comment by amirad** [comment]
Yessss @ipsha21 I'm thinking of ways that mentors can gain credibility on the platform and showcase how they have supported people. It will be a good incentive to be a mentor and they will feel appreciated after.
